#!/usr/bin/env python
PACKAGE="image_manip"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("frame_rate", double_t, 1, "frame rate Hz", 20, 0.1, 50)
gen.add("phi", double_t, 1, "phi", 0, -3.2, 3.2)
gen.add("theta", double_t, 1, "theta", 0, -0.05, 0.05)
gen.add("psi", double_t, 1, "psi", 0, -0.05, 0.05)
gen.add("center_x", double_t, 1, "center_x", 0, -0.5, 0.5)
gen.add("center_y", double_t, 1, "center_y", 0, -0.5, 0.5)
# gen.add("center_z", double_t, 1, "center_z", 0, -0.5, 0.5)
gen.add("off_x", double_t, 1, "off_x", 0, -0.5, 0.5)
gen.add("off_y", double_t, 1, "off_y", 0, -0.5, 0.5)
gen.add("off_z", double_t, 1, "off_z", 0, 0.0, 0.3)
# gen.add("z", double_t, 1, "z", 1.0, 0.1, 10.0)

exit(gen.generate(PACKAGE, "image_manip", "RotoZoom"))
